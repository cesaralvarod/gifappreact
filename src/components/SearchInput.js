import React, { useState } from "react";
import PropTypes from "prop-types";

// Styles

import "./SearchInput.css";

const SearchInput = ({ handleQuery }) => {
  const [query, setQuery] = useState("");

  const handleSubmit = (e) => {
    e.preventDefault();

    if (query !== "") {
      handleQuery(query);
    }
  };

  const handleInputChange = (e) => {
    setQuery(e.target.value);
  };

  return (
    <>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          className="input-search"
          name="q"
          id="query"
          placeholder="Search a gif"
          autoComplete="off"
          value={query}
          onChange={handleInputChange}
        />
      </form>
    </>
  );
};

SearchInput.propTypes = {
  handleQuery: PropTypes.func.isRequired,
};

export default SearchInput;
