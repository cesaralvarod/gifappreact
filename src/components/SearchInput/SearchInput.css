input.input-search {
  background: transparent;
  border: none;
  border: 2px solid #eb7709;
  box-sizing: border-box;
  font-size: 1.6rem;
  padding: 0.8rem 1rem;
  width: 100%;
}

input.input-search:focus {
  outline: none;
}

input.input-search:focus-within {
  border: none;
  border: 2px solid #eb7209;
}
