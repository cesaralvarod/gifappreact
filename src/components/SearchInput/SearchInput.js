import React from "react";

// Styles

import "./SearchInput.css";

function SearchInput(props) {
  const handleSubmit = (e) => {
    e.preventDefault();
  };

  const handleInputChange = (e) => {
    props.handleQuery(e.target.value);
  };

  return (
    <>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          className="input-search"
          name="q"
          id="query"
          placeholder="Search a gif"
          autoComplete="off"
          onInput={handleInputChange}
        />
      </form>
    </>
  );
}

export default SearchInput;
