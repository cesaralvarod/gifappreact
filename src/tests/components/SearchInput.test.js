import React from "react";
import { shallow } from "enzyme";
import SearchInput from "../../components/SearchInput";

describe("Testing in SearchInput.js", () => {
  // let funcParam = () => {};
  let handleQuery = jest.fn();

  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks(); // Clear mock or simulation for something

    wrapper = shallow(<SearchInput handleQuery={handleQuery} />);
  });

  test("Component <SearchInput /> should show correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Simulate input event from component <Searchinput />", () => {
    const input = wrapper.find("input");

    const value = "Rick and Morty";

    input.simulate("change", { target: { value } } /* this is my event */);

    expect(input.props().value).toBe(value);
  });

  test("Shouldn't post info when form is submit", () => {
    const form = wrapper.find("form");

    form.simulate("submit", { preventDefault() {} });

    expect(handleQuery).not.toHaveBeenCalled();
  });

  test("Should call handleQuery() and clean input text", () => {
    const form = wrapper.find("form");

    const input = wrapper.find("input");

    const value = "Rick and Morty";

    input.simulate("change", { target: { value } });

    form.simulate("submit", { preventDefault() {} });

    expect(handleQuery).toHaveBeenCalled();
  });
});
