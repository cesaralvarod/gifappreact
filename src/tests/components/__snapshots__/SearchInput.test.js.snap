// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in SearchInput.js Component <SearchInput /> should show correctly 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="input-search"
      id="query"
      name="q"
      onChange={[Function]}
      placeholder="Search a gif"
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
